src/wallet/test/coinselector_tests.cpp

```diff
@@ -344,6 +344,73 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
         const auto result10 = SelectCoins(*wallet, coins, 10 * CENT, coin_control, coin_selection_params_bnb);
         BOOST_CHECK(result10);
     }
+    {
+        // single coin should be selected when effective fee > long term fee
+        std::unique_ptr<CWallet> wallet = std::make_unique<CWallet>(m_node.chain.get(), "", m_args, CreateMockWalletDatabase());
+        wallet->LoadWallet();
+        LOCK(wallet->cs_wallet);
+        wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+        wallet->SetupDescriptorScriptPubKeyMans();
+
+        std::vector<COutput> coins;
+
+        expected_result.Clear();
+        add_coin(coins, *wallet, 10 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 9 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 1 * CENT, 6 * 24, false, 0, true);
+        add_coin(10 * CENT, 2, expected_result);
+        CCoinControl coin_control;
+        coin_selection_params_bnb.m_effective_feerate = CFeeRate(5000);
+        coin_selection_params_bnb.m_long_term_feerate = CFeeRate(3000);
+        const auto result11 = SelectCoins(*wallet, coins, 10 * CENT, coin_control, coin_selection_params_bnb);
+        BOOST_CHECK(EquivalentResult(expected_result, *result11));
+    }
+    {
+        // more coins should be selected when effective fee < long term fee
+        std::unique_ptr<CWallet> wallet = std::make_unique<CWallet>(m_node.chain.get(), "", m_args, CreateMockWalletDatabase());
+        wallet->LoadWallet();
+        LOCK(wallet->cs_wallet);
+        wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+        wallet->SetupDescriptorScriptPubKeyMans();
+
+        std::vector<COutput> coins;
+
+        expected_result.Clear();
+        add_coin(coins, *wallet, 10 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 9 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 1 * CENT, 6 * 24, false, 0, true);
```

Comment:
In 5c15850ded2a3195fae0dd238e1aebaaa46b13ce "3 new tests for SelectCoins function"

All of this setup is the same in all three test cases. These can be deduplicated and just have a single scope for all these cases.

Reply:
made these changes.